src/classes/VOL_CTRL_VolunteersJobListingFS.cls

```diff
@@ -301,12 +302,29 @@ global virtual with sharing class VOL_CTRL_VolunteersJobListingFS {
                 }
                 
                 // now remove any jobs whose shifts are all past date.
-                for (integer i = listVolunteerJobs.size() - 1; i >= 0; i--) {
-                    Volunteer_Job__c job = listVolunteerJobs[i];
-                    if (job.Number_of_Shifts__c > 0 && job.Volunteer_Job_Slots__r.size() == 0) {
-                        listVolunteerJobs.remove(i);                    
-                    }
+                // unless the user clicked on a specific event from the calendar
+                if (shiftIdFilter == null && jobIdFilter == null) {
+                	for (integer i = listVolunteerJobs.size() - 1; i >= 0; i--) {
+	                    Volunteer_Job__c job = listVolunteerJobs[i];
+	                    if (job.Number_of_Shifts__c > 0 && job.Volunteer_Job_Slots__r.size() == 0) {
+	                        listVolunteerJobs.remove(i);                    
+	                    }
+	                }
+                }
+                
+                // if one event was selected; check if the event was in the past
+                eventInThePast = false;
+                if (listVolunteerJobs.size() == 1) {
+                	Volunteer_Job__c job = listVolunteerJobs[0];
+                	eventInThePast = 0 < job.Number_of_Shifts__c;
+                	for (Volunteer_Shift__c shift : job.Volunteer_Job_Slots__r) {
+                		if (date.today() <= shift.Start_Date_Time__c) {
```

Comment:
seems like this logic will say the job is an event in the past, even if it has multiple shifts, some of which are old and some of which are new.  am I misunderstanding this?